examples/embeddable_examples/public/react_embeddables/field_list/field_list_embeddable.tsx

```diff
@@ -0,0 +1,243 @@
+import { FieldListApi, Services, FieldListSerializedState, FieldListRuntimeState } from './types';
+
+const DataViewPicker = withSuspense(LazyDataViewPicker, null);
+
+const defaultFieldListState: WithAllKeys<FieldListRuntimeState> = {
+  dataViewId: undefined,
+  selectedFieldNames: undefined,
+  dataViews: undefined,
+};
+
+const getCreationOptions: UnifiedFieldListSidebarContainerProps['getCreationOptions'] = () => {
+  return {
+    originatingApp: '',
+    localStorageKeyPrefix: 'examples',
+    timeRangeUpdatesType: 'timefilter',
+    compressed: true,
+    showSidebarToggleButton: false,
+    disablePopularFields: true,
+  };
+};
+
+const deserializeState = async (
+  dataViews: DataViewsPublicPluginStart,
+  serializedState?: SerializedPanelState<FieldListSerializedState>
+): Promise<FieldListRuntimeState> => {
+  const state = serializedState?.rawState ? cloneDeep(serializedState?.rawState) : {};
+  // inject the reference
+  const dataViewIdRef = (serializedState?.references ?? []).find(
+    (ref) => ref.name === FIELD_LIST_DATA_VIEW_REF_NAME
+  );
+  // if the serializedState already contains a dataViewId, we don't want to overwrite it. (Unsaved state can cause this)
+  if (dataViewIdRef && state && !state.dataViewId) {
+    state.dataViewId = dataViewIdRef?.id;
+  }
+
+  const [allDataViews, defaultDataViewId] = await Promise.all([
+    dataViews.getIdsWithTitle(),
+    dataViews.getDefaultId(),
+  ]);
+  if (!defaultDataViewId || allDataViews.length === 0) {
+    throw new Error(
+      i18n.translate('embeddableExamples.unifiedFieldList.noDefaultDataViewErrorMessage', {
+        defaultMessage: 'The field list must be used with at least one Data View present',
+      })
+    );
+  }
+  const initialDataViewId = state.dataViewId ?? defaultDataViewId;
+  const initialDataView = await dataViews.get(initialDataViewId);
+  return {
+    dataViewId: initialDataViewId,
+    selectedFieldNames: state.selectedFieldNames ?? [],
+    dataViews: [initialDataView],
+  };
+};
+
+export const getFieldListFactory = (
+  core: CoreStart,
+  { dataViews, data, charts, fieldFormats }: Services
+) => {
+  const fieldListEmbeddableFactory: EmbeddableFactory<FieldListSerializedState, FieldListApi> = {
+    type: FIELD_LIST_ID,
+    buildEmbeddable: async ({ initialState, finalizeApi, parentApi, uuid }) => {
+      const state = await deserializeState(dataViews, initialState);
+      const allDataViews = await dataViews.getIdsWithTitle();
+      const subscriptions = new Subscription();
+      const titleManager = initializeTitleManager(initialState?.rawState ?? {});
+      const fieldListStateManager = initializeStateManager(state, defaultFieldListState);
+
+      // Whenever the data view changes, we want to update the data views and reset the selectedFields in the field list state manager.
+      subscriptions.add(
+        fieldListStateManager.api.dataViewId$
+          .pipe(
+            skip(1),
+            distinctUntilChanged(),
```

Comment:
Why was `distinctUntilChanged` added? `fieldListStateManager.api.dataViewId$.next` getting called with identical values? You should not need  `distinctUntilChanged` here, and instead ensure `fieldListStateManager.api.dataViewId$.next` is only called when there is a new value.

Reply:
I was trying a fix a bug where Reset would wipe out the `selectedFieldNames`, but that bug exists on `main` and is probably outside the scope of this effort. 


https://github.com/user-attachments/assets/1ba4c959-8842-4a09-acb9-37a32c88e77e

Reply:
reverted in [8a485b1](https://github.com/elastic/kibana/pull/218174/commits/8a485b1c70c5c6b7978cdbf5af4edd4dcd5cd60f)